shard-operator-parent/shard-operator-core/src/test/java/com/redhat/service/smartevents/shard/operator/core/utils/DeploymentStatusUtilsTest.java

```diff
@@ -0,0 +1,40 @@
+package com.redhat.service.smartevents.shard.operator.core.utils;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import com.redhat.service.smartevents.shard.operator.core.TemplateProvider;
+
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.api.model.apps.DeploymentCondition;
+
+public class DeploymentStatusUtilsTest {
+
+    @Test
+    public void TestIsTimeoutFailure_success() {
+        Deployment deployment = TemplateProvider.loadTestDeployment();
+        DeploymentCondition timeoutCondition = new DeploymentCondition();
+        timeoutCondition.setType("Progressing");
+        timeoutCondition.setStatus("False");
+        timeoutCondition.setReason("ProgressDeadlineExceeded");
+        deployment.getStatus().getConditions().add(timeoutCondition);
+        boolean result = DeploymentStatusUtils.isTimeoutFailure(deployment);
+        Assertions.assertThat(result).isTrue();
```

Comment:
Minor comment, you can merge in a single statement (same comment for other assertions)
```suggestion
        Assertions.assertThat(DeploymentStatusUtils.isTimeoutFailure(deployment))
            .isTrue();
```